x-pack/plugins/visualization_editor/public/index.tsx

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { I18nProvider } from '@kbn/i18n/react';
+import React from 'react';
+import { render, unmountComponentAtNode } from 'react-dom';
+import chrome from 'ui/chrome';
+import { PLUGIN_ID } from '../common';
+
+// TODO: Convert this to the "new platform" way of doing UI
```

Comment:
Is there something more we can already prepare for the new platform here? There is the shimming concept, but I don't know whether that's important for client side code... ping @lukeelmers

Reply:
In the new platform, we will not have an App for this anyway- it will be embedded

Reply:
yeah if this isn't going to be an app you might not need to worry too much.

but in general, to make life easier on yourself as you keep adding to this plugin, i'd still consider setting up a new platform-style plugin definition from the outset, with inter-plugin dependencies shimmed in.

check out [the architecture section](https://github.com/elastic/kibana/blob/master/src/core/MIGRATION.md#architecture) of the migration overview document for an idea of what i mean. or [this pr](https://github.com/elastic/kibana/pull/35625) i'm working on for the plugin that will house vis infrastructure.

basically you could do something like this:
- add a `public/plugin.ts` file with `setup` and `stop` methods (probably just skip the constructor for now)
- instantiate the plugin definition from `public/index.ts` (this is your shim - call `new Plugin().setup(...shimmedDependencies)`)
- then in your plugin.setup method, then kick off all the rest of your services from there, passing the dependencies down to _those_ services' setup methods.

honestly, you are welcome to set this up however you want, it just depends on how much work you are willing to do later. my strongest recommendation would be to build the plugin in such a way that stateful imports are all done at the top level, and passed down. this will save you a **lot** of work down the road. 

happy to sync if you want to chat about any of this

Reply:
Thanks @lukeelmers , this is what I meant. The vis PR is very helpful. @wylieconlon no strong opinion on this, but I would suggest we use this set up from the start. I can also do this in a separate PR